processor/deltatocumulativeprocessor/internal/data/add.go

```diff
@@ -24,9 +25,44 @@ func (dp Number) Add(in Number) Number {
 	return dp
 }
 
-// nolint
 func (dp Histogram) Add(in Histogram) Histogram {
-	panic("todo")
+	// bounds different: no way to merge, so reset observation to new boundaries
+	if !pslice.Equal(dp.ExplicitBounds(), in.ExplicitBounds()) {
+		in.MoveTo(dp.HistogramDataPoint)
+		return dp
+	}
+
```

Comment:
Do we need to do the `StartTimeUnixNano` checks like we do for Sums?

Reply:
we are! the code I think you talk about is in `delta.Accumulator[T]`, which applies to all datatypes

https://github.com/sh0rez/opentelemetry-collector-contrib/blob/57d7906d430f6ccd1e4eda0493a1fb1c7d8cf590/processor/deltatocumulativeprocessor/internal/delta/delta.go#L38-L45

Reply:
Ahhh. that's why I missed it :)